sources/ClangSharp/Cursors/Decls/AccessSpecDecl.cs

```diff
@@ -0,0 +1,11 @@
+using ClangSharp.Interop;
+
+namespace ClangSharp
+{
+    public sealed class AccessSpecDecl : Decl
+    {
+        internal AccessSpecDecl(CXCursor handle) : base(handle, CXCursorKind.CXCursor_CXXAccessSpecifier)
```

Comment:
Most of the types were copy pasted and had the accessibility of the constructor fixed up.

The `Debug.Assert(handle.Kind == <expectedKind>)` calls were changed to pass the expected kind down to the base constructor.